tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBrokerParcelable.java

```diff
@@ -37,13 +57,91 @@ protected void cache() {
+            }
+        });
+        Observable.fromArray(this.accounts)
+                .observeOn(AndroidSchedulers.mainThread(), true)
+                .subscribeOn(Schedulers.io())
+                .flatMapIterable(new Function<List<TradeItLinkedBrokerAccountParcelable>, Iterable<TradeItLinkedBrokerAccountParcelable>>() {
+                    @Override
+                    public Iterable<TradeItLinkedBrokerAccountParcelable> apply(@NonNull List<TradeItLinkedBrokerAccountParcelable> linkedBrokerAccountParcelables) throws Exception {
+                        return linkedBrokerAccountParcelables;
+                    }
+                }).map(new Function<TradeItLinkedBrokerAccountParcelable, Single<TradeItLinkedBrokerAccountParcelable>>() {
+                    @Override
+                    public Single<TradeItLinkedBrokerAccountParcelable> apply(@NonNull final TradeItLinkedBrokerAccountParcelable linkedBrokerAccountParcelable) throws Exception {
+                        return Single.create(new SingleOnSubscribe<TradeItLinkedBrokerAccountParcelable>() {
+                            @Override
+                            public void subscribe(@NonNull final SingleEmitter<TradeItLinkedBrokerAccountParcelable> emmiter) throws Exception {
+                                linkedBrokerAccountParcelable.refreshBalance(new TradeItCallback<TradeItBalanceParcelable>() {
+                                    @Override
+                                    public void onSuccess(TradeItBalanceParcelable balance) {
+                                        emmiter.onSuccess(linkedBrokerAccountParcelable);
+                                    }
+
+                                    @Override
+                                    public void onError(TradeItErrorResult error) {
+                                        Log.e(TAG, error.toString());
+                                        emmiter.onError(new RuntimeException(error.toString()));
+                                    }
+                                });
+                            }
+                        });
+                    }
+                }).flatMap(new Function<Single<TradeItLinkedBrokerAccountParcelable>, ObservableSource<?>>() {
+                        @Override
+                        public ObservableSource<?> apply(@NonNull Single<TradeItLinkedBrokerAccountParcelable> tradeItLinkedBrokerAccountParcelableSingle) throws Exception {
+                            return tradeItLinkedBrokerAccountParcelableSingle.toObservable();
+                        }
+                }).subscribe(new DisposableObserver() {
+                    @Override
+                    public void onNext(@NonNull Object linkedBrokerAccountParcelable) {
+                        Log.d(TAG, "onNext: " + linkedBrokerAccountParcelable);
+                    }
+
+                    @Override
+                    public void onError(@NonNull Throwable e) {
+                        callback.onFinished();
```

Comment:
So I tested it. It is called if at least one error occurred and when all linkedBrokers have been processed. To summarize it will be called only once at the end.